src/state_machine.zig

```diff
@@ -932,17 +1126,62 @@ pub fn StateMachineType(
             defer self.scan_result_count = 0;
 
             const result_size: usize = self.scan_result_count * @sizeOf(Transfer);
-            const results: []Transfer = mem.bytesAsSlice(Transfer, output[0..result_size]);
             stdx.copy_disjoint(
                 .exact,
-                Transfer,
-                results,
-                self.scan_buffer[0..self.scan_result_count],
+                u8,
+                output[0..result_size],
+                self.scan_buffer[0..result_size],
             );
 
             return result_size;
         }
 
+        fn execute_get_account_history(
+            self: *StateMachine,
+            input: []const u8,
+            output: *align(16) [constants.message_body_size_max]u8,
+        ) usize {
+            if (self.scan_result_count == 0) return 0;
+            defer self.scan_result_count = 0;
+
+            const filter: AccountFilter = mem.bytesToValue(
+                AccountFilter,
+                input[0..@sizeOf(AccountFilter)],
+            );
+
+            const scan_results: []const AccountHistoryGrooveValue = mem.bytesAsSlice(
+                AccountHistoryGrooveValue,
+                self.scan_buffer[0 .. self.scan_result_count * @sizeOf(AccountHistoryGrooveValue)],
+            );
+
+            const output_slice: []AccountBalance = mem.bytesAsSlice(AccountBalance, output);
+            var output_count: usize = 0;
+
+            for (scan_results) |*result| {
```

Comment:
We could `assert((filter.account_id == result.dr_account_id) != (filter.account_id == result.cr_account_id))`.. to verify that _exactly_ one of the account ids matches. (Right now the unreachable checks that _at least_ one of the ids matches.)

Then again we have a pretty strong guarantee about this on the "create accounts" side so maybe its not helpful.